src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContentOfT.cs

```diff
@@ -1,112 +1,41 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-#if !NETCOREAPP
-using System.Diagnostics;
-#endif
 using System.IO;
-using System.Text;
+using System.Net.Http.Headers;
 using System.Text.Json;
 using System.Text.Json.Serialization.Metadata;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace System.Net.Http.Json
 {
-    internal sealed partial class JsonContent<TValue> : HttpContent
+    internal sealed class JsonContent<TValue> : JsonContent
```

Comment:
I suspect it might be possible to do away with the generic `JsonContent<T>` and always use the weakly typed variant. If the motivation is to avoid boxing `_typedValue`, I suspect this could result in that happening more than once as the `ValueCore` property is being evaluated.

Reply:
Yes, that could be done, and yes my concern was boxing. My theory was that the primarily used path is serialization, where we could avoid boxing, and the calling `Value` was much less likely so having boxing there was okay. But let me know if you disagree and would prefer to avoid `JsonContent<T>` altogether.

Reply:
I don't think it's hugely important all things considered. It's probably going to be worse anyway if ValueCore is called multiple times.

Reply:
Okay, makes sense to me. I imagine `JsonContent` is used by reference types the vast majority of the time anyway. I've pushed the changes, it definitely makes things simpler.